### DISCUSSION TITLE
Conditional matching rules

### DISCUSSION THREAD

**Discussion by ludorival** [pact-foundation/pact-js#811] [Q&A]
Hi,

I try to implement Pact in my organisation but I face an issue when I try to verify the contract. I need your inputs on what could be the right way to approach when a request returns a list of mixed elements.
 For example, let's say we have a list of payment means. We can have a Credit Card, a Paypal account or a Wire Transfer account.
Each payment mean is identified by a payment method `CREDIT_CARD, PAYPAL, WIRE_TRANSFER`.
Each payment mean has some specific fields. A credit card is identified by a credit card number, and expiry date, a card type and a card holder name whereas a Paypal account has just the email and the WireTransfer, the IBAN. So our request can returns something like this 
```json
[
    {
        "paymentMethod" : "CREDIT_CARD",
        "cardNumber" : "4444*****4448",
        "cardType" : "VISA",
        "expiryDate" : "2024/01",
        "cardHolderName" : "John Doe"
    },
    {
        "paymentMethod" : "PAYPAL",
        "email": "john.doe@test.com"
    },
    {
        "paymentMethod" : "WIRE_TRANSFER",
        "iban": "NL82INGB3659362247"
    }
]
```
I know we can tackle this issue by expecting to have the same value returned by the provider but if it is not the case. The provider can return different combinaison of value.
How I can define that in my contract ?

**Comment by ludorival** [comment]
I just saw in the Pact V4, we have the [`combine`](https://github.com/pact-foundation/pact-specification/tree/version-4#matching-rules) matcher, let me try with this rule.

**Reply by mefellows** [reply]
The `combine` matcher is not supported in Pact JS at this time.

There is an `arrayContaining` matcher which might address your need, available in the current beta: https://github.com/pact-foundation/pact-js/tree/feat/v3.0.0#array-contains-matcher

Just take the usual forewarning that it's a beta, and things will likely change.
